public/app.d.ts

```diff
@@ -1706,42 +1706,505 @@ declare namespace JMap {
+       * 
+       * If there is no active form, do nothing.
+       * 
+       * @example ```ts
+       * 
+       * // clear the current query form data
+       * JMap.Application.Print.clearDefaultData()
+       * ```
+       */
       function clearDefaultData(): void
+
+      /**
+       * **JMap.Application.Query.displayInDialog**
+       *
+       * By default the query form is displayed in the left panel with a simple layout, one field per line.
+       * 
+       * But it is possible to display it in a modal window, with the layout defined in the administration.
+       * 
+       * @throws if isVisibleInDialog is not a boolean
+       * @param isVisibleInDialog true in modal window, false in left panel
+       * @example ```ts
+       * 
+       * // display the current form in a modal window
+       * JMap.Application.Print.displayInDialog(true)
+       * 
+       * // display the current form in the left panel
+       * JMap.Application.Print.displayInDialog(false)
+       * ```
+       */
       function displayInDialog(isVisibleInDialog: boolean): void
-      function processQuery(values: any): Promise<void>
+
+      /**
+       * **JMap.Application.Query.processQuery**
+       *
+       * Process active query form with the given values.  
+       * 
+       * Throws if no query is active.
+       * 
+       * Resolve only if some features are found.
+       * 
+       * If features are found, we make the layer selectable and visible if needed, then select and zoom on found features.
+       * 
+       * If no feature are found, reject promise. When rejected, it rejects human readable error, using user locale.
+       * 
+       * @throws if no query is active
+       * @param value the form data, depends on the form
+       * @example ```ts
+       * 
+       * // display the current form in a modal window
+       * JMap.Application.Print
+       *   .processQuery({
+       *    $param1: "black",
+       *    $param2: "white"
+       *   })
+       *   .then(features => console.info(`Found ${features.length} feature(s)`, features))
+       *   .catch(error => {
+       *     // pass here if no features are found, or other error
+       *     // thrown error is a readable error using the user locale
+       *     console.error(error)
+       *   })
+       * ```
+       */
+      function processQuery(values: any): Promise<GeoJSON.Feature[]>
     }
 
+    /**
+     * **JMap.Application.Event**
+     * 
+     * From this section you can manage your own event listeners reacting to NG App library events.
+     * 
+     * For all your listener you need provide a listener id. We introduced this notion of listener ids in order
+     * to be able to know what's the problem if something goes wrong in a listener.
+     * 
+     * Like that we are able to identify more easily the problem in the javascript console.
+     * 
+     * Listener ids have to be unique for the namespace, regardless to the type of event you register your listener.
```

Comment:
"..... , regardless to the type of event you register your listener to."

(ajouter "to" à la fin)